hippod/report_generator.py

```diff
@@ -166,18 +197,22 @@ def convert(self, app):
                 title = item[3]
                 files_catalog[sub_dir]['title'] = title
                 subcontainer = os.path.join(db_path, sha_major[0:2], sha_major, sha_minor, 'subcontainer.db')
-                if achievement_id != '':
-                    achievement = os.path.join(db_path, sha_major[0:2], sha_major, sha_minor, 'achievements', '{}.db'.format(achievement_id))
-                    with open(achievement, 'r') as achiev:
-                        content = json.load(achiev)
-                    stored_data_path = ReportGenerator.ReportGeneratorDocument.store_achievement(app, content, sub_dir)
-                    files_catalog[sub_dir]['achievement'] = stored_data_path
+                achievement = os.path.join(db_path, sha_major[0:2], sha_major, sha_minor, 'achievements', '{}.db'.format(achievement_id))
+                with open(achievement, 'r') as achiev:
+                    content = json.load(achiev)
+                stored_data_path = ReportGenerator.ReportGeneratorDocument.store_achievement(app, content, sub_dir)
+                files_catalog[sub_dir]['achievement'] = stored_data_path
                 with open(subcontainer, 'r') as subc:
                     content = json.load(subc)
+                if 'data' not in content['object-item']:
+                    continue
                 data_list = content['object-item']['data']
                 for i, data in enumerate(data_list):
                     stored_data_path = ReportGenerator.ReportGeneratorDocument.store_data(app, data, sub_dir)
+                    if stored_data_path == None:
+                        continue
                     files_catalog[sub_dir]['data'].append(stored_data_path)
+            print('here files catalog {}'.format(files_catalog))
```

Comment:
The whole "for j, item in enumerate(self.list_of_lists)" loop is somewhat long and hard to understand. Move it into a separate method and call the method accordingly. If you have problem because the method do so much different things you should not move it into one function but in several separate function doing **one** thing.

This should always the case. Moving larger groups of code into separate functions and name then accordingly.